code/modules/mob/living/simple_animal/hostile/giant_spider.dm

```diff
@@ -304,6 +343,92 @@
 		busy = 0
 		stop_automated_movement = 0
 
+/mob/living/simple_animal/hostile/giant_spider/verb/web()
+	set name = "Spin Web"
+	set desc = "Create a web that slows down movement."
+	set category = "Greimorian"
+
+	var/obj/effect/spider/stickyweb/W = locate() in get_turf(src)
+	if(!W)
+		to_chat(usr, SPAN_NOTICE("\The [src] begins to secrete a sticky substance."))
+		if(!do_after(src, 20)) return
+		new /obj/effect/spider/stickyweb(src.loc)
+
+
+/mob/living/simple_animal/hostile/giant_spider/nurse/verb/cocoon()
+	set name = "Cocoon and Feed"
+	set desc = "Cocoon an incapacitated mob so you can feed upon it. This will give you one food point."
+	set category = "Greimorian"
+
+
+	var/list/available_mobs = list()
+
+	for(var/mob/living/A in range(1, src))
+		if(A.stat && !istype(A,/mob/living/simple_animal/hostile/giant_spider) && !A.isSynthetic())
+			available_mobs += A
+	var/mob/P = tgui_input_list(usr, "Choose a mob to cocoon.", "Cocoon", available_mobs)
+	if(get_dist(src, P) <= 1)
+		src.visible_message("\The [src] begins to secrete a sticky substance around \the [P].")
+		if(!do_after(src, 80)) return
+		if(P && istype(P.loc, /turf) && get_dist(src,P) <= 1)
+			var/obj/effect/spider/cocoon/C = new(P.loc)
+			var/large_cocoon = FALSE
+			C.pixel_x = P.pixel_x
+			C.pixel_y = P.pixel_y
+			for(P in C.loc)
+				if(istype(P, /mob/living/simple_animal/hostile/giant_spider))
+					continue
+				large_cocoon = TRUE
+				fed++
+				src.visible_message("\The [src] sticks a proboscis into \the [P] and sucks a viscous substance out.")
+				P.forceMove(C)
+				C.pixel_x = P.pixel_x
+				C.pixel_y = P.pixel_y
+				break
+			for(var/obj/item/I in C.loc)
+				I.forceMove(C)
+			for(var/obj/structure/S in C.loc)
+				if(!S.anchored)
+					S.forceMove(C)
+				large_cocoon = TRUE
+			for(var/obj/machinery/M in C.loc)
+				if(!M.anchored)
+					M.forceMove(C)
+				large_cocoon = TRUE
```

Comment:
These repeats a list traversal when it looks like they have no need to:

- Traverse the list once
- Use the type check to select what to do

eg:

```
for(P in C.loc)
  if(istype(P, /mob/living/simple_animal/hostile/giant_spider)
    continue
  if(istype(P, /obj/item)
    var/obj/item/I = P
    P.forceMove(C)
etc. etc.
```

Reply:
Handled